source/tests/test.ts

```diff
@@ -1,147 +1,246 @@
+import m from '..'; // tslint:disable-line:import-name
 
 const isNode8orHigher = Number(process.versions.node.split('.')[0]) >= 8;
 
-const PromiseSubclassFixture = class extends Promise {};
-const ErrorSubclassFixture = class extends Error {};
+// Currently out of order, see https://github.com/Microsoft/TypeScript/issues/15202 class PromiseSubclassFixture<T> extends Promise<T> {}
+class ErrorSubclassFixture extends Error {}
 
 const document = jsdom();
-const createDomElement = el => document.createElement(el);
+const createDomElement = (el: string) => document.createElement(el);
 
-const types = new Map([
-	['undefined', undefined],
-	['null', null],
-	['string', [
-		'🦄',
-		'hello world',
-		''
-	]],
-	['number', [
-		6,
-		1.4,
-		0,
-		-0,
-		Infinity,
-		-Infinity
-	]],
-	['boolean', [
-		true,
-		false
-	]],
-	['symbol', Symbol('🦄')],
-	['array', [
-		[1, 2],
-		new Array(2)
-	]],
-	['function', [
-		function foo() {}, // eslint-disable-line func-names
-		function () {},
-		() => {},
-		async function () {},
-		function * () {}
-	]],
-	['buffer', Buffer.from('🦄')],
-	['object', [
-		{x: 1},
-		Object.create({x: 1})
-	]],
-	['regExp', [
-		/\w/,
-		new RegExp('\\w')
-	]],
-	['date', new Date()],
-	['error', [
-		new Error('🦄'),
-		new ErrorSubclassFixture()
-	]],
-	['nativePromise', [
-		Promise.resolve(),
-		PromiseSubclassFixture.resolve()
-	]],
-	['promise', {then() {}, catch() {}}],
-	['generator', (function * () {
-		yield 4;
-	})()],
-	['generatorFunction', function * () {
-		yield 4;
-	}],
-	['asyncFunction', [
-		async function () {},
-		async () => {}
-	]],
-	['map', new Map()],
-	['set', new Set()],
-	['weakMap', new WeakMap()],
-	['int8Array', new Int8Array()],
-	['uint8Array', new Uint8Array()],
-	['uint8ClampedArray', new Uint8ClampedArray()],
-	['uint16Array', new Uint16Array()],
-	['int32Array', new Int32Array()],
-	['uint32Array', new Uint32Array()],
-	['float32Array', new Float32Array()],
-	['float64Array', new Float64Array()],
-	['arrayBuffer', new ArrayBuffer(10)],
-	['nan', [
-		NaN,
-		Number.NaN
-	]],
-	['nullOrUndefined', [
-		null,
-		undefined
-	]],
-	['plainObject', [
-		{x: 1},
-		Object.create(null),
-		new Object() // eslint-disable-line no-new-object
-	]],
-	['integer', 6],
-	['safeInteger', [
-		Math.pow(2, 53) - 1,
-		-Math.pow(2, 53) + 1
-	]],
-	['domElement', [
-		'div',
-		'input',
-		'span',
-		'img',
-		'canvas',
-		'script'
-	].map(createDomElement)],
-	['non-domElements', [
-		document.createTextNode('data'),
-		document.createProcessingInstruction('xml-stylesheet', 'href="mycss.css" type="text/css"'),
-		document.createComment('This is a comment'),
-		document,
-		document.implementation.createDocumentType('svg:svg', '-//W3C//DTD SVG 1.1//EN', 'http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd'),
-		document.createDocumentFragment()
-	]],
-	['infinite', [
-		Infinity,
-		-Infinity
-	]]
+interface Test {
+	is(value: any): boolean;
+	fixtures: any[];
+}
+
+const types = new Map<string, Test>([
```

Comment:
In terms of readability, I would rather see it formatted like this

```ts
const types = new Map<string, Test>([
	['undefined', {
		is: m.undefined,
		fixtures: [
			undefined
		]
	}],
	['null', {
		is: m.null_,
		fixtures: [
			null
		]
	}],
	['string', {
		is: m.string,
		fixtures: [
			'🦄',
			'hello world',
			''
		]
	}]
	// ...
]);
```

Reply:
Agreed